### ISSUE TITLE
Turn off friendly fire

### ISSUE THREAD

**Issue by jeremygurr** [jeremygurr/dcssca#361]
This eliminates so many spoilery special cases that already exist in the game, wipes out some annoying situations, and opens up new strategic options. 

It's a pretty big buff, but let's see how it fares with all the other nerfs happening. If it ends up too powerful, we can weaken things in a few different ways. But I think friendly fire is just unnecessary. Like item destruction. It's logical, but not fun.

**Comment by Sandman25DCSS** [comment]
Do you mean an Ice Elementalist will be able to throw icicles through Ice Beasts?

**Comment by jeremygurr** [comment]
exactly.

**Comment by SleepySteel** [comment]
I think that it should be a thing with some projectiles but not, say, firestorm. Making everything able to pass through followers also feels like it makes polearm weapons a bit less special.

**Comment by Sandman25DCSS** [comment]
Yes, I like friendly fire too. It's interesting to decide "Should I create deadly summons or will I need fan/slouch/apocalypse/disaster area/etc. later?"

**Comment by jeremygurr** [comment]
I agree with the value of the strategic decision as it applies to all area effects. Area effects should damage friendlies, and so should be carefully considered. 

But I think single targetting should work through friendlies, just like battle sphere does now, and just like a polearm can reach through a friendly. I think the same advantage should be given to magic missles, bows and slings, etc.

**Comment by Sandman25DCSS** [comment]
It is not only strategic decisions, it helps to differentiate spells and abilities. For example, Venom Mage can happily join Yred to use Venom Bolt to shoot through allied zombies unlike Fire Elementalist who cannot do it with Fireball or Bolt of Fire. Airstrike deals less damage than it could because it is smite-targeted, launchers require Portal Projectile to make sure Summoned Butterflies are not OP etc. With suggested changes you can use even Summon Mammal to be almost immune to melee in corridors: shoot through the mammal if it was lucky to survive previous monster attacks, summon a new one otherwise. Lightning Spire in Elf corridor would no longer require Freezing/Poisonous Cloud to safely kill banishers etc.

**Comment by jeremygurr** [comment]
Why can a venom mage shoot through undead minions but not a ranged attacker? This is spoilery, inconsistent, and unintuitive, things I'm trying hard to reduce here. 

I actually understand why fire bolt can't. And I agree with that, because firebolt is a penetrating spell, designed to damage everything it passes through, like lightning bolt. But unless an arrow has the penetration brand, it should also be able to pass through friendlies like venom bolt can. 

So penetrating and area effects still hurt friendlies. But single target attacks should pass through just fine. I think lightning spire has a penetrating attack (although maybe it doesn't have to?), so it also would damage friendlies it passed through. Battlesphere does not, and so it's current behavior of shooting through enemies makes sense to me. 

This makes portal projectile pointless, but I'm not sure that's a bad thing. I find it tedious to use as a general strategy, and I don't think there's a good enough case to cripple some ranged attacks / spells over others (since there are already many exceptions in the game where pass-through-friendlies already happens). 

Of course the same rules should apply to monster attacks, and in fact I think it already works that way in many cases. 

This is not the same as smite targetting, however. If you summon a horde of butterlies, since they aren't friendly to an enemy archer, that enemy must shoot them first before getting through to the player.

**Comment by Sandman25DCSS** [comment]
> Why can a venom mage shoot through undead minions but not a ranged attacker? This is spoilery, inconsistent, and unintuitive, things I'm trying hard to reduce here. 

Ranged attacker with penetrating brand can do it too, just don't expect that intelligent allies will remain friendly :)
I don't see any spoilers here. Venom Bolt still can hit allies too and when it does it loses one range but continues further (Bolt of Fire/Cold/Lignting/Draining work the same way except Lightning does not lose range). Hitting undead allies with Venom Bolt is not a big deal, they get exactly 0 damage (because Venom Bolt does no damage to monsters who are immune to poison) and they don't turn hostile of course.

> But unless an arrow has the penetration brand, it should also be able to pass through friendlies like venom bolt can. 

As I wrote above, it is already possible except those allies will turn hostile as expected. It is not a spoiler that allies become hostile when intentionally damaged.

> But single target attacks should pass through just fine.

I disagree. When single target attack hits friendly monsters and continues further, it is not a single target attack. Also the hit ally should become hostile (to avoid spoilers like "You can hit your ally with longbow but not with long sword").

> since there are already many exceptions in the game where pass-through-friendlies already happens.

Could you elaborate on that? Maybe I will learn something new.

> This is not the same as smite targetting, however. If you summon a horde of butterlies, since they aren't friendly to an enemy archer, that enemy must shoot them first before getting through to the player. 

Yes, I know, it's another reason why I don't support the idea, I dislike player buffs for no reason.

**Comment by Sandman25DCSS** [comment]
Of course you can use Bolt of Draining to shoot through zombies too but it's obvious probably.

**Comment by Sandman25DCSS** [comment]
> since there are already many exceptions in the game where pass-through-friendlies already happens

To be specific I know only one exception - Fedhas. And it is ok/fun, that's the power of gods, they are supposed to provide unique experience.

**Comment by jeremygurr** [comment]
the other exceptions I can think of are battlesphere (which often shoots through friendlies without hurting them),  cleaving axe attacks, and polearm reaching (which conveniently pass right through friendlies). 

Do you think cleaving axe attacks or polearm reaches would be more fun if they also damaged friendlies? Or that battlesphere should not attack if there isn't a clear shot? 

That would force different gameplay decisions and limit options in the same way that ranged attackers or conjurers are prevented from having summons or worshipping yred or combining their skills with necromancy now. 

What do we lose if single hit shots were allowed to go through friendlies? The ranged attacker suddenly gets the option to use summons, which definitey makes them more powerful, but no more than an axe wielder that can have summons, or a battle sphere user, or any other player that branches out in different skills. It comes with a natural limiter (training two skills takes much longer than training one).

But this might still be too great a buff for ranged attackers, so I propose this counter nerf: ranged attackers can't attack at melee range. They have to use a melee weapon, a summon, a spell, or a wand. Still plenty of options, but they are weakened by being forced to hybridize a little. 

Now if we went down this path, I would want another change to make fighting as a ranged character less tedious. The big problem here is the constant weapon swapping as monsters get too close. So instead, I think after ammo is removed the "quiver" slot would be used to arm a ranged weapon. So a player would have a melee and a ranged weapon armed at the same time. It's no different than having large rocks ready to throw while wielding a sword. When they hit tab, and the monster is far, the ranged weapon is fired. If the monster is too close, the melee weapon is used, without any swap penalty. Not realistic, but possibly more fun to play.

**Comment by yrmvgh** [comment]
I'm generally in favor of fixing "the friendly fire problem" as it is simulationist, annoying, and makes some combos impossible in a game where almost everything is possible... albeit many possible things are "inadviseable".

My solution of choice to the friendly fire problem is for friendly fire to be ignored by allies (shit happens, war is hell, sorry buddy but I'll buy you a beer when this fight is over) or to make them "back up" but not immediately desire to turn and kill you. Maybe they disappear, maybe they turn neutral, maybe they just decide "for the next 10 rounds, I'm going to stay 3 or more tiles away from all enemies and allies since it's so dangerous in there", but they shouldn't treat every accidental rock or splash as a declaration of war.

But there's already perfectly good code in there for "shoot through friends" so I fully support simply making use of it instead of creating a whole new system. Possibly for the simulationists out there, we can throw a bone like "-3 acc, -3% damage, and -3% on the MR check when shooting / magicking through allies" but only if turning off friendly fire actually proves to be broken or too easy.

**Comment by Sandman25DCSS** [comment]
Thank you, I learned a new thing indeed, I didn't know about Battlesphere and always avoided allies to let it have a line of fire.
Personally I don't care about realism so I don't see any issues with cleaving or polearms (you can use polearm through hostile monster too by the way though it still has a chance to fail and hit the adjacent monster instead).
I wouldn't like forcing hybridization of ranged characters, this is common complaint in vanilla crawl and personally I enjoy pure ranged characters too. Obligatory Okawaru is not a big deal as it is a great good for ranged characters even without missile gifts.

**Comment by jeremygurr** [comment]
An interesting idea was suggested which might resolve some of the pain here. If a summon or friendly gets hit by friendly fire, they just leave (disappear). This allows Ds with demonic guardian, for example, to still cast area effect spells, even when the guardian pops up uninvited. There should be no other penalty.

**Comment by yrmvgh** [comment]
> If a summon or friendly gets hit by friendly fire, they just leave (disappear).

.

> **Maybe they disappear**, maybe they turn neutral, maybe they just decide "for the next 10 rounds, I'm going to stay 3 or more tiles away from all enemies and allies since it's so dangerous in there", but they shouldn't treat every accidental rock or splash as a declaration of war.

I kind of already said that, but perhaps I proposed too many ideas. :wink: I'm glad it's gaining traction though! FF ftl!

**Comment by jeremygurr** [comment]
I just found out that the game already allows friendly fire through demonic guardians. I'm going to try making that apply to all friendlies. I'm wondering if the tediousness removed plus the increased options of combining summoning/necromancy with ranged attacks is worth more than the strategic loss that comes from not having to decide whether to be a ranged attacker or a summoner. 

It's a simple quick change that can easily be taken back if we really don't like it.
